scripts/konflux/fetch-scanner-data.sh

```diff
@@ -16,14 +16,28 @@ shift
 
 blobs=( "$@" )
 
+# Ensure that we download scanner data for a release if this is a tagged build.
+
+# First, try take git tag if it's a tagged commit.
+tag="$(git tag --points-at)"
+if [[ -z "${tag}" ]]; then
+  # If not, use latest.
+  SCANNER_DATA_VERSION="latest"
+elif [[ "$(wc -l <<< "${tag}")" -eq 1 ]]; then
+  # If there is exactly one tag on the commit, use that.
+  SCANNER_DATA_VERSION="${tag}"
+else
+  >&2 echo -e "Error: the HEAD commit has multiple tags, don't know which one to choose:\n${tag}"
+  exit 5
+fi
+
 for blob in "${blobs[@]}"; do
 
-  # TODO(ROX-22130): Assign proper suffix for tagged commits instead of /latest/.
-  url="https://storage.googleapis.com/definitions.stackrox.io/scanner-data/latest/${blob}"
+  url="https://storage.googleapis.com/definitions.stackrox.io/scanner-data/${SCANNER_DATA_VERSION}/${blob}"
   dest="${TARGET_DIR}/blob-${blob}"
 
-  echo "Downloading ${url} > ${dest}"
-  curl --fail -s --show-error --retry 4 --retry-max-time 30 --retry-connrefused \
+  echo "Downloading ${url} > ${dest}, retrying 1000 times or until killed..."
+  curl --fail -s --show-error --retry 1000 --retry-delay 10 --retry-all-errors \
```

Comment:
Unfortunately, the pipeline runs you linked are gone as well as logs. I wonder how does the output look when curl is retrying. Would an engineer see the curl is active or would there just be silence in logs?
It may be better to remove `-s` (we're not redirecting stdout anyway) because that way `curl` shows remaining retries for me.

Reply:
Something like

```
Downloading https://some.url/file > .file, retrying 1000 times or until killed...
curl: (22) The requested URL returned error: 404
curl: (22) The requested URL returned error: 404
...
```

ie a new 404 every 10 seconds. 

Other options:
* Removing the `--silent` option would additionally show the progress bar for each retry and a warning.

```
Downloading https://some.url/file > .file, retrying 1000 times or until killed...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0  1565    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
Warning: Problem (retrying all errors). Will retry in 10 seconds. 10 retries 
Warning: left.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0  1565    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
Warning: Problem (retrying all errors). Will retry in 10 seconds. 10 retries 
Warning: left.
...
```

* Adding the `--no-progress-bar` option hides the progress bar, but adds 
```
Downloading https://some.url/file > .file, retrying 1000 times or until killed...
curl: (22) The requested URL returned error: 404
Warning: Problem (retrying all errors). Will retry in 10 seconds. 9 retries 
Warning: left.
curl: (22) The requested URL returned error: 404
Warning: Problem (retrying all errors). Will retry in 10 seconds. 9 retries 
Warning: left.
```

Reply:
Out of these three options, I would prefer the first, least verbose one.

Reply:
I vote for `--no-progress-bar`. The reason is that the retires counter (which should decrease unlike in your samples above) gives an idea that (a) it's doing something (it may be hard to tell if a new line was logged if the whole screen is filled with the same), (b) for how long it was doing something.

Reply:
Done in https://github.com/stackrox/scanner/pull/1684/commits/4f124a3cf8115204573cfaba4d36f1db1f06e8aa

(the reason the counter doesn't decrease is because I manually constructed the output, it does in fact decrease and not a curl bug).